pkg/scheduler/plugins/elasticquota/plugin_helper.go

```diff
@@ -48,17 +47,27 @@ func (g *Plugin) getPodAssociateQuotaName(pod *v1.Pod) string {
 	return quotaName
 }
 
-var GetQuotaName = func(pod *v1.Pod, quotaLister schedulinglisterv1alpha1.ElasticQuotaLister) string {
+func (g *Plugin) GetQuotaName(pod *v1.Pod) string {
 	quotaName := extension.GetQuotaName(pod)
 	if quotaName != "" {
 		return quotaName
 	}
-	eq, err := quotaLister.ElasticQuotas(pod.Namespace).Get(pod.Namespace)
+	eq, err := g.quotaLister.ElasticQuotas(pod.Namespace).Get(pod.Namespace)
 	if err == nil && eq != nil {
 		return eq.Name
 	} else if !errors.IsNotFound(err) {
 		klog.Errorf("Failed to Get ElasticQuota %s, err: %v", pod.Namespace, err)
 	}
+
+	eqList, err := g.quotaInformer.GetIndexer().ByIndex("annotation.namespaces", pod.Namespace)
+	if err != nil {
+		return extension.DefaultQuotaName
+	}
+
+	if len(eqList) > 0 {
+		return pod.Namespace
```

Comment:
For example, there is an ElasticQuota named `eq-a`, and limited namespaces `default`, `kube-system` and `other`. Now there is new Pod in namespace `default`, and no any ElasticQuotas in namespace `default`,  so the Pod's ElasticQuota is `eq-a`..  As the result, the returned quota name is `eq-a` not `default`. Am I right?

Reply:
fixed